logs_api/lib/opentelemetry/logs/logger.rb

```diff
@@ -12,36 +12,45 @@ class Logger
 
       # Emit a {LogRecord} to the processing pipeline.
       #
-      # @param timestamp [optional Float, Time] Time in nanoseconds since Unix
+      # @param [optional Float, Time] timestamp Time in nanoseconds since Unix
       #   epoch when the event occurred measured by the origin clock, i.e. the
       #   time at the source.
-      # @param observed_timestamp [optional Float, Time] Time in nanoseconds
+      # @param [optional Float, Time] observed_timestamp Time in nanoseconds
       #   since Unix epoch when the event was observed by the collection system.
       #   Intended default: Process.clock_gettime(Process::CLOCK_REALTIME)
```

Comment:
The units are unclear here. If we accept nanoseconds, I'd expect that to be Integer rather than Float, and if we accept `Time`, then the units are not relevant. IMO we should only accept `Time`, and we should convert that to "Integer nanoseconds since Epoch" internally.

Related: documenting the `Process.clock_gettime(Process::CLOCK_REALTIME)` as the default increases confusion about the units (and precision).

Reply:
Great points, I agree. Addressed here: https://github.com/open-telemetry/opentelemetry-ruby/pull/1611#discussion_r1678505898